template/web/src/components/contract-alert/ContractAlert.tsx

```diff
@@ -0,0 +1,58 @@
+import { ExclamationTriangleIcon } from '@radix-ui/react-icons';
+import { Abi, parseEther } from 'viem';
+
+import { UseContractReturn } from '../../hooks/contracts';
+import { useLoggedInUserCanAfford } from '../../hooks/useUserCanAfford';
+
+function useCanUserAfford(amount: number) {
+  return useLoggedInUserCanAfford(parseEther(String(amount)));
+}
+
+function ContractAlertLayout({ children }: { children: React.ReactNode }) {
+  return (
+    <div className="my-3 flex items-center justify-center">
+      <div className="mr-2">
+        <ExclamationTriangleIcon width={12} height={12} />
+      </div>
+      <div className="text-xs">{children}</div>
+    </div>
+  );
+}
+
+type ContractAlertProps = {
+  contract: UseContractReturn<Abi>;
+  amount: number;
+};
+
+export default function ContractAlert({ contract, amount }: ContractAlertProps) {
```

Comment:
Are we building a Toast?

Reply:
Not a proper Toast in this PR. My plan was to improve this after cleaning up buy-me-coffee

Reply:
Just FYI, `main` has to always be in a good state. So is this PR going to make the code half baked?